experimental/dds/tree2/api-report/tree2.api.md

```diff
@@ -1023,7 +1035,10 @@ interface Invariant<in out T> extends Contravariant<T>, Covariant<T> {
 }
 
 // @alpha
-type isAny<T> = boolean extends (T extends {} ? true : false) ? true : false;
+type isAny<T> = boolean extends (T extends never ? true : false) ? true : false;
```

Comment:
the old version of this incorrectly detected types which were a union of undefined and some other type as any. This new check fixes that, and seems to be more robust. Tests have been added to prevent regression in this way.